src/south/SQLDbToFile/SQLDbToFile.class.js

```diff
@@ -91,81 +96,100 @@ class SQLDbToFile extends ProtocolHandler {
+   * write them into a CSV file and send to the Engine.
+   * @param {String} scanMode - The scan mode
+   * @param {Date} startTime - The start time
+   * @param {Date} endTime - The end time
    * @return {void}
    */
-  async onScanImplementation(scanMode) {
+  async historyQuery(scanMode, startTime, endTime) {
     if (!this.timezone) {
       this.logger.error('Invalid timezone')
       return
     }
 
-    if (this.ongoingReads[scanMode]) {
-      this.logger.silly(`onScan ignored: ongoingReads[${scanMode}]: ${this.ongoingReads[scanMode]}`)
-      return
-    }
-
     this.ongoingReads[scanMode] = true
 
     let result = []
-    try {
-      switch (this.driver) {
-        case 'mssql':
-          result = await this.getDataFromMSSQL(this.lastCompletedAt[scanMode])
-          break
-        case 'mysql':
-          result = await this.getDataFromMySQL(this.lastCompletedAt[scanMode])
-          break
-        case 'postgresql':
-          result = await this.getDataFromPostgreSQL(this.lastCompletedAt[scanMode])
-          break
-        case 'oracle':
-          result = await this.getDataFromOracle(this.lastCompletedAt[scanMode])
-          break
-        case 'sqlite':
-          result = await this.getDataFromSqlite(this.lastCompletedAt[scanMode])
-          break
-        default:
-          this.logger.error(`Driver ${this.driver} not supported by ${this.dataSource.dataSourceId}`)
-          result = []
+    let intervalResult = []
+    do {
+      try {
+        this.logQuery(this.query, startTime, endTime, this.maxReturnValues)
+
+        switch (this.driver) {
+          case 'mssql':
+            // eslint-disable-next-line no-await-in-loop
+            intervalResult = await this.getDataFromMSSQL(startTime, endTime)
+            break
+          case 'mysql':
+            // eslint-disable-next-line no-await-in-loop
+            intervalResult = await this.getDataFromMySQL(startTime, endTime)
+            break
+          case 'postgresql':
+            // eslint-disable-next-line no-await-in-loop
+            intervalResult = await this.getDataFromPostgreSQL(startTime, endTime)
+            break
+          case 'oracle':
+            // eslint-disable-next-line no-await-in-loop
+            intervalResult = await this.getDataFromOracle(startTime, endTime)
+            break
+          case 'sqlite':
+            // eslint-disable-next-line no-await-in-loop
+            intervalResult = await this.getDataFromSqlite(startTime, endTime)
+            break
+          default:
+            this.logger.error(`Driver ${this.driver} not supported by ${this.dataSource.dataSourceId}`)
+        }
+        result = result.concat(intervalResult)
+        // eslint-disable-next-line no-param-reassign
+        startTime = this.getLatestDate(result, startTime)
+      } catch (error) {
+        this.logger.error(error)
       }
-    } catch (error) {
-      this.logger.error(error)
-    }
+    } while (intervalResult.length)
```

Comment:
Instead of this check, wouldn't it be better to check intervalResult.length AND the last value/timestamp returned from the previous intervalResult ? If both are the same, it will leave the while loop and avoid the problem you saw in PostgreSQL